addons/LPCAnimatedSprite/LPCSpriteSheet.gd

```diff
@@ -47,47 +52,110 @@ var NormalAnimationData:Array[LPCAnimationData] = [
 ]
-var ThrustAnimationData:Array[LPCAnimationData] = [
+
+static var ThrustAnimationData:Array[LPCAnimationData] = [
 	LPCAnimationData.new(6,"THRUST_UP",0, 0,false),
 	LPCAnimationData.new(6,"THRUST_LEFT",1, 0,false),
 	LPCAnimationData.new(6,"THRUST_DOWN",2, 0,false),
 	LPCAnimationData.new(6,"THRUST_RIGHT",3, 0,false),
 ]
-var RodAnimationData:Array[LPCAnimationData] = [
+static var RodAnimationData:Array[LPCAnimationData] = [
 	LPCAnimationData.new(6,"ROD_UP",0, 0,false),
 	LPCAnimationData.new(6,"ROD_LEFT",1, 0,false),
 	LPCAnimationData.new(6,"ROD_DOWN",2, 0,false),
 	LPCAnimationData.new(6,"ROD_RIGHT",3, 0,false),
 ]
-var WhipAnimationData:Array[LPCAnimationData] = [
+static var WhipAnimationData:Array[LPCAnimationData] = [
 	LPCAnimationData.new(8,"WHIP_UP",0, 0,false),
 	LPCAnimationData.new(8,"WHIP_LEFT",1, 0,false),
 	LPCAnimationData.new(8,"WHIP_DOWN",2, 0,false),
 	LPCAnimationData.new(8,"WHIP_RIGHT",3, 0,false),
 ]
 
-func Size() -> int:
-	match SpriteType:
-		SpriteTypeEnum.Normal:
-			return 64
-		_:
-			return 192
+enum SpritesheetType
+{
+	SizeNormal,
+	Size_6_6_8
+}
+
+func GetSpritesheetType():
+	if SpriteSheet.get_height() == 3648:
+		return SpritesheetType.Size_6_6_8
+	return SpritesheetType.SizeNormal
+	
+func GetSpriteRect(animationData:LPCAnimationData, frame: int) -> Rect2:
+	const startOfOversizedAnimation = 21;
+	var spriteSize:int = animationData.Size
+	match GetSpritesheetType():
+		LPCSpriteSheet.SpritesheetType.SizeNormal:
+			return Rect2(spriteSize*(frame+animationData.Col), spriteSize*animationData.Row, spriteSize, spriteSize)
+		LPCSpriteSheet.SpritesheetType.Size_6_6_8:
+			if animationData.Row < startOfOversizedAnimation:
+				return Rect2(spriteSize*(frame+animationData.Col), spriteSize*animationData.Row, spriteSize, spriteSize)
+			else:
+				var startPosition:int = startOfOversizedAnimation * 64
+				var row = animationData.Row - startOfOversizedAnimation
+				return Rect2(spriteSize*(frame+animationData.Col), startPosition + spriteSize*row, spriteSize, spriteSize)
+	return Rect2(spriteSize*(frame+animationData.Col), spriteSize*animationData.Row, spriteSize, spriteSize)
+	
 func AnimationData() -> Array[LPCAnimationData]:
 	match SpriteType:
 		SpriteTypeEnum.Normal:
-			return NormalAnimationData
+			match GetSpritesheetType():
+				SpritesheetType.SizeNormal:
+					return NormalAnimationData
+				SpritesheetType.Size_6_6_8:
+					var animationData:Array[LPCAnimationData] = []
+					animationData.append_array(NormalAnimationData)
+					animationData.append_array(PrecompiledOversizedSlash)
+					animationData.append_array(PrecompiledOversizedSlashReverse)
+					animationData.append_array(PrecompiledOversizedThrust)
+					return animationData
+				_:
+					return NormalAnimationData
 		SpriteTypeEnum.OversizeRod:
 			return RodAnimationData
 		SpriteTypeEnum.OversizeThrust:
 			return ThrustAnimationData
 		SpriteTypeEnum.OversizeSlash:
 			return SlashAnimationData
+		SpriteTypeEnum.OversizeSlashReverse:
```

Comment:
I don't see how a precompiled sheet with an oversize animation will gets it no-oversized walk, etc. animations with this. Don't we still need a means to detect precompiled on top of the enum?

Reply:
I'll merge this PR and in the example project you can see all of it working